File: Pattern/Binary Tree Traversal/257. Binary Tree Paths/Solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def binaryTreePaths(self, root: Optional[TreeNode]) -> List[str]:
        result = []
        self.helper(root,'',result)
        return result
    def helper(self,node,path,result):
        if node is None:
            return ;
        if len(path) > 0 :
            path+='->'
        path+= str(node.val)
        if node.left == None and node.right == None :
            result.append(path)
        else :
            self.helper(node.left,path,result)
            self.helper(node.right,path,result)

File: pattern/Binary Tree Traversal/257. Binary Tree Paths/Solution.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {string[]}
 */
var binaryTreePaths = function (root) {
  let path = "";
  let result = [];
  helper(root, path, result);
  return result;
};

function helper(node, path, result) {
  path += node.val;
  if (node.left == null && node.right == null) {
    result.push(path);
    return;
  }
  path += "->";
  if (node.left) {
    helper(node.left, path, result);
  }
  if (node.right) {
    helper(node.right, path, result);
  }
}
